### PR TITLE
BRANCHES.md: Add explanation of branches and using forks

### PR DESCRIPTION
**Pull request by kewillford**
Adding a file to explain the branches in the repo and how we are using them as well as other helpful hints with forking.

### CODE REVIEW

File: BRANCHES.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+Branches used in this repo
+==========================
+
+The document explains the branching structure that we are using in the git for VFSForGit repository as well as the forking strategy that we have adopted for contributing.
```
Comment (benpeart) [inline BRANCHES.md]: "we are using in the git for VFSForGit" is a bit awkward.  How about "we are using in the VFSForGit repository"

Comment (kewillford) [reply BRANCHES.md]: Yeah that will sound better.

File: BRANCHES.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+==========================
+
+The document explains the branching structure that we are using in the git for VFSForGit repository as well as the forking strategy that we have adopted for contributing.
+
+Repo Branches
+-------------
+
+1. master
+    
+    This will track the git for windows repository master branch
```
Comment (derrickstolee) [inline BRANCHES.md]: "Git for Windows"

File: BRANCHES.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+2. gvfs
```
Comment (derrickstolee) [inline BRANCHES.md]: Why have we not renamed this to 'vfs' or something?

Comment (kewillford) [reply BRANCHES.md]: Haven't gotten around to renaming these.  There are builds that will need to be updated.  There are tags and releases that are using 'gvfs'.  Not sure if we would want to change those, especially the releases.  The version for git is determined using the tags. @dscho, I'm sure you have automation around these branch names and tags, how much work is it to change this?  We could add the new names and have both while we get things updated.

Comment (dscho) [reply BRANCHES.md]: > I'm sure you have automation around these branch names and tags, how much work is it to change this?

I think the most important automation around this is the `GIT-VERSION-GEN` patches: febd85092824fdd716b5b9a9515ab721ed64dbeb and f357366f470ab397c72232721c9a4eef2268d698

Other than that: there are *plenty* of references in our internal builds. However, I think they mostly refer to the internal `gvfs/master` branch. Most of those internal builds are obsolete now, I think, e.g. the job that used to tell me when I need to synchronize the clean patches with the internal state, and the job that used to create and publish the installers (which I think we still want, but have to re-do?).

In short: there is not much automation that needs adjusting, so I would suggest changing it and seeing what breaks 😉
